xcc/settings.py

```diff
@@ -87,15 +87,45 @@ class Config:  # pylint: disable=missing-class-docstring
         env_file = get_path_to_env_file()
         env_prefix = get_name_of_env_var()
 
+    def _sanity_check(self, key: str, val: str) -> None:
+        """
+        Check for conditions that make saving the env_file
+        dangerous to the user.
+
+        Parameters
+        ----------
+        key : str
+            env file key
+        val : str
+            env file value
+
+        Raises
+        ------
+        Exception
+            When the value should not be saved to the env_file.
+
+        Returns
+        -------
+        None
+            Has no effect if all is safe.
+
+        """
+        # Unprintables in REFRESH_TOKEN assumed never to be valid (0x0a, 0x20, etc.)
+
+        if key == "REFRESH_TOKEN" and val is not None and not val.isprintable():
+            raise Exception("REFRESH_TOKEN contains non-printable character(s)")
```

Comment:
Usually, it is bad practice to raise an `Exception` when a more specific subclass will do:
```suggestion
            raise ValueError("REFRESH_TOKEN contains non-printable character(s)")
```
Also, to match the style of the error messages in [commands.py](https://github.com/XanaduAI/xanadu-cloud-client/blob/main/xcc/commands.py#L76), we would phrase this as:
```suggestion
            raise ValueError("The REFRESH_TOKEN setting contains non-printable character(s).")
```

Reply:
Did this in the code